README.md

```diff
@@ -295,10 +295,11 @@ configPaths: ['./.jsonschema-tools.yaml']
 # that they are at least within these bounds.
 enforcedNumericBounds: [Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER],
 
-# If true, repository tests for schemas will fail if the schema doesn't have
-# examples.  If it does have examples, they will be also tested for
-# validation against the schema.
-requireExamples: false
+# If true an example will be generated during schema materialization.
+# Examples already present in the schema will be preserved. E.g. If the
+# schema already has examples, and shouldGenerateExample is true,
+# no new example will be generated.
+shouldGenerateExample: false
```

Comment:
This kind of gets into software design philosophy (configurability vs. just doing the Right Thing), but should this really be mutually exclusive with `requireExamples`?  I can imagine someone wanting both to require examples and to require schema implementers to create them by hand.

(Also, where is `requireExamples` enforced? The only place I can find it is here in the README file.)

Reply:
I removed it in a previous patch, but forgot to remove it from the README at that time.

This `shouldGenerateExamples` flag covers the same use case that `requireExamples` does for us, so I removed `requireExamples` as an option.  Setting `shouldGenerateExample` will ensure that every schema has an example.  I could see someone wanting to require examples without generating them, but let's wait until that is a needed use case.

Also, `requireExamples` was only enforced in repo tests, not by any materialization logic.  If someone wasn't using the repo tests, `requireExamples` wouldn't have any effect.  I'd actually like to refactor some of the way we apply these tests.  Right now it is difficult, if not impossible, to choose which tests you'd like to apply.  I've got three categories of repo tests, and it is possible to hack around which ones you run, but for the individual test cases, there's no way to selectively choose them.  Ideally the tests would work more like a linter, with easy configuration of what lint rules to enforce.  By not including `requireExamples` as a regular config...we won't have to figure out how to move it into a future lint rules config if we ever are able to create them.